fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/TableRowSource.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright IBM Corp. 2019,2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query;
+
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.query.expression.StatementRenderer;
+
+/**
+ * Represents a table referenced in the from list
+ */
+public class TableRowSource implements RowSource {
+    // optional schema name
+    private final String schemaName;
+
+    // The name of the table
+    private final String tableName;
+
+    /**
+     * Protected constructor for a table without a qualified schema name
+     *
+     * @param tableName
+     */
+    protected TableRowSource(String tableName) {
+        DataDefinitionUtil.assertValidName(tableName);
+        this.schemaName = null;
+        this.tableName  = tableName;
+    }
+
+    @Override
+    public Alias getImpliedAlias() {
+        // use the tableName as the alias (used when a real alias is not given)
+        return new Alias(this.tableName);
+    }
+
+    /**
+     * Protected constructor for a table qualified by schema name
+     *
+     * @param schemaName
+     * @param tableName
+     * @param alias
+     */
+    protected TableRowSource(String schemaName, String tableName) {
+        DataDefinitionUtil.assertValidName(schemaName);
+        DataDefinitionUtil.assertValidName(tableName);
```

Comment:
Is this necessary since rowSource line 258 in StringStatementRender.java does this check in getQualifiedName

Reply:
Still useful if this class gets used in a different context

Reply:
seems like a good defense